lnst/Recipes/ENRT/SimpleNetnsRouterRecipe.py

```diff
@@ -0,0 +1,95 @@
+from lnst.Recipes.ENRT.helpers import ip_endpoint_pairs
+from lnst.Controller.NetNamespace import NetNamespace
+from lnst.Common.IpAddress import interface_addresses
+from lnst.Devices.VethPair import VethPair
+from collections.abc import Collection
+
+class SimpleNetnsRouterRecipe(SimpleNetworkRecipe):
+    """
+    This recipe implements Enrt testing for a simple network scenario that
+    looks as follows
+
+    .. code-block:: none
+
+                    +--------+
+             +------+ switch +-----+
+             |      +--------+     |
+          +--+-+                 +-+--+
+        +-|eth0|-+          +----|eth0|-----+
+        | +----+ |          |    +--+-+     |
+        | host1  |          | host2 |       |
+        +--------+          |     +-+--+    |
+                            |   +-|veth|--+ |
+                            |   | +----+  | |
+                            |   | host2ns | |
+                            |   +---------+ |
+                            +---------------+
+
+    This is a derived class from SimpleNetworkRecipe which creates a netns on
+    the second host, enables forwarding into it and returns it as second
+    endpoint. Perf and ping tests will therefore excercise the second host's
+    routing path.
+    """
+    netns_ipv4 = IPv4NetworkParam(default="192.168.102.0/24")
+    netns_ipv6 = IPv6NetworkParam(default="fc00:1::/64")
+
+    def test_wide_configuration(self, config):
+        """
+        Test wide configuration for this recipe consists of:
+        - Creating the netns on host2, named 'host2ns'
+        - Adding a veth pair pn0 and np0, the latter sits inside the netns
+          - host2.pn0    = 192.168.102.1/24 and fc00:1::1/64
+          - host2.ns.np0 = 192.168.102.2/24 and fc00:1::2/64
+        - Add routes inside the netns and on host1
+        - Set addr_gen_mode=0 sysctl on both hosts, avoiding issues with
+          router solicitations
+        """
+        host1 = self.matched.host1
+        host2 = self.matched.host2
+        config = super().test_wide_configuration(config)
+
+        host2.ns = NetNamespace("host2ns")
+
+        host2.pn0, host2.np0 = VethPair()
+        host2.ns.np0 = host2.np0
+
+        ipv4_addr = interface_addresses(self.params.netns_ipv4)
+        ipv6_addr = interface_addresses(self.params.netns_ipv6)
+        config.configure_and_track_ip(host2.pn0, next(ipv4_addr))
+        config.configure_and_track_ip(host2.pn0, next(ipv6_addr))
+        config.configure_and_track_ip(host2.ns.np0, next(ipv4_addr))
+        config.configure_and_track_ip(host2.ns.np0, next(ipv6_addr))
+        host2.pn0.up_and_wait()
+        host2.ns.np0.up_and_wait()
+
+        for gw in (self.params.netns_ipv4[1], self.params.netns_ipv6[1]):
+            host2.ns.np0._ipr_wrapper("route", "add", dst="default",
+                                      gateway=f"{gw}")
+
+        host2.run("sysctl -w net.ipv4.ip_forward=1")
+        host2.run("sysctl -w net.ipv6.conf.all.forwarding=1")
```

Comment:
unless these are default config values, i think these need the reverse action added to the `test_wide_deconfiguration` method, otherwise they may affect other tests running in sequence after this one.

we used to have a tester api for `sysctl` that used to take care of this automatically but we don't have this right now...

Reply:
this applies to other calls of `host.run('sysctl...')` as well...

Reply:
Ah, thanks! I missed that. Do I have to kill the netns as well? If so, will `host2.ns = None` do? (Please pardon my ignorance.)

Reply:
namespaces are cleaned up automatically so just the sysctls will do, thanks!